### PR TITLE
Apply pipeline to db api endpoint

### PR DESCRIPTION
**Pull request by delgermurun**
refs #5508
- adds pipeline to export, import and delete all methods

### CODE REVIEW

**Comment by ErisDS** [comment]
I think that it should be possible to further refactor this to use `utils.handlePermissions(docName, methodName)` rather than having your own version?

**Comment by delgermurun** [comment]
@ErisDS updated. Thanks for the guide.

File: core/server/api/db.js [hunk 1]
```diff
@@ -51,31 +59,42 @@ db = {
...
         options = options || {};
 
-        // Check if a file was provided
-        if (!utils.checkFileExists(options, 'importfile')) {
-            return Promise.reject(new errors.NoPermissionError('Please select a file to import.'));
-        }
+        function validate(options) {
+            // Check if a file was provided
+            if (!utils.checkFileExists(options, 'importfile')) {
+                return Promise.reject(new errors.NoPermissionError('Please select a file to import.'));
```
Comment (delgermurun) [inline core/server/api/db.js]: @ErisDS I guess those two 2 errors should be ValidationError. If so I'll send PR on it.

Comment (ErisDS) [inline core/server/api/db.js]: Yes, good point :+1:

**Pull request merged**